src/ast/mod.rs

```diff
@@ -3650,6 +3653,8 @@ impl fmt::Display for CloseCursor {
 pub struct Function {
     pub name: ObjectName,
     pub args: Vec<FunctionArg>,
+    // e.g. `x > 5` in `COUNT(x) FILTER (WHERE x > 5)`
+    pub filter: Option<Box<Expr>>,
```

Comment:
The current way to represent aggregates with filters (in the select clause) is https://docs.rs/sqlparser/latest/sqlparser/ast/enum.Expr.html#variant.AggregateExpressionWithFilter

The idea of adding the `filter` directly to the Function I think is more elegant and general

What do you think about (in some other PR) removing `AggregateExpressionWithFilter` and just using this more general formulation?

Reply:
Yes, that is a good idea :+1:

Reply:
But we would have to make sure that supporting things like `(x + y) filter (where z)` was actually accidental